plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/AbstractMetastoreTableOperations.java

```diff
@@ -89,7 +89,7 @@ protected final String getRefreshedLocation(boolean invalidateCaches)
     @Override
     protected final void commitNewTable(TableMetadata metadata)
     {
-        String newMetadataLocation = writeNewMetadata(metadata, version + 1);
+        String newMetadataLocation = writeNewMetadata(metadata, version.orElse(-1) + 1);
```

Comment:
I prefer the Glue approach for `commitNewTable`, personally.
- Verify `version.isEmpty()`
- Hardcode the version number to 0

Reply:
Addressed!